drivers/docker/driver_test.go

```diff
@@ -378,6 +378,49 @@ func TestDockerDriver_Start_StoppedContainer(t *testing.T) {
 	require.NoError(t, d.DestroyTask(task.ID, true))
 }
 
+// TestDockerDriver_ContainerAlreadyExists asserts that when Nomad tries to start
+// a job and the container already exists, it purges it, and starts it again (as
+// opposed to trying to continuously re-create an already existing container)
+//
+// See https://github.com/hashicorp/nomad/issues/22218
+func TestDockerDriver_ContainerAlreadyExists(t *testing.T) {
+	ci.Parallel(t)
+	testutil.DockerCompatible(t)
+
+	task, cfg, _ := dockerTask(t)
+	must.NoError(t, task.EncodeConcreteDriverConfig(cfg))
+
+	client := newTestDockerClient(t)
+	driver := dockerDriverHarness(t, nil)
+	cleanup := driver.MkAllocDir(task, true)
+	defer cleanup()
+	copyImage(t, task.TaskDir(), "busybox.tar")
+
+	d, ok := driver.Impl().(*Driver)
+	must.True(t, ok)
+
+	_, err := d.createImage(task, cfg, client)
+	must.NoError(t, err)
+
+	containerCfg, err := d.createContainerConfig(task, cfg, cfg.Image)
+	must.NoError(t, err)
+
+	// create and start a container
```

Comment:
Right. I also made another unit test which starts the container but this case is covered by another unit test, and it's a very easy one because when the container is running the method just returns very quickly. The comment here is wrong of course, I'll adjust.

Reply:
fixed in bfe9f85